planner/planner_APIs.go

```diff
@@ -338,7 +339,7 @@ func (planner *MyPlanner) Planning(ctx context.Context, planningRequest *solutio
 		}
 		travelPlan.ID = topSolution.ID
 		resp.TravelPlans[sIdx] = travelPlan
-		resp.TripDetailsURL[sIdx] = "/v1/plans/" + travelPlan.ID
+		resp.TripDetailsURL[sIdx] = "/v1/plans/" + travelPlan.ID + "?" + "date=" + planningRequest.TravelDate
```

Comment:
@weihesdlegend I modify the redirect URL on the plan show button, so it also includes the travel date, which allows us to save it directly on the plan show page. 
To do so, I have to use the new field `TravelDate` added in `planningRequest`. Let me know what you think.

Reply:
I think it's ok, but is there handling when the `TravelDate` and the plan do not match?

Reply:
I see, let me think and check this situation.

Reply:
Here I don't think there could be a mismatch between `TravelDate` and plan, because `TravelDate` is from user search input as part of the `planningRequest`. The plan solution should match with the search input. Let me know what you think.

Reply:
That's true but if user tampering with the URL on the result page then they may get unexpected results